### PR TITLE
Ks approve fix

### PR DESCRIPTION
**Pull request by captnseagraves**
This PR includes changes from https://github.com/gitcoinco/web/pull/3116 and addresses https://github.com/gitcoinco/web/issues/3069

### CODE REVIEW

File: app/grants/models.py [hunk 1]
```diff
@@ -302,6 +307,21 @@ class Subscription(SuperModel):
         max_digits=50,
         help_text=_('The required gas price for the Subscription.'),
     )
+    sub_new_approve_tx_id = models.CharField(
```
Comment (mbeacom) [inline app/grants/models.py]: Can we omit `sub_` from these fields?

File: app/grants/views.py [hunk 2]
```diff
@@ -306,6 +307,7 @@ def grant_fund(request, grant_id,  grant_slug):
         subscription.token_address = request.POST.get('denomination', '')
         subscription.token_symbol = request.POST.get('token_symbol', '')
         subscription.gas_price = request.POST.get('gas_price', 0)
+        subscription.sub_new_approve_tx_id = request.POST.get('sub_new_approve_tx_id', '')
```
Comment (mbeacom) [inline app/grants/views.py]: We really need to migrate this to a form or something 🤔

File: app/grants/models.py [hunk 3]
```diff
@@ -515,10 +535,10 @@ class ContributionQuerySet(models.QuerySet):
 class Contribution(SuperModel):
     """Define the structure of a subscription agreement."""
 
-    tx_id = models.CharField(
+    contribution_tx_id = models.CharField(
```
[the remark this answers is not in the document]
Comment (captnseagraves) [reply app/grants/models.py]: The goal is to have descriptive naming conventions. I suppose they are redundant.

File: app/grants/models.py [hunk 4]
```diff
@@ -515,10 +535,10 @@ class ContributionQuerySet(models.QuerySet):
...
         max_length=255,
         default='0x0',
-        help_text=_('The transaction ID of the Contribution.'),
+        help_text=_('The transaction id for the contribution.'),
```
[the remark this answers is not in the document]
Comment (captnseagraves) [reply app/grants/models.py]: This field was rewritten. I can change the description back.

File: app/grants/views.py (hunk 2 above)
Comment (captnseagraves) [reply app/grants/views.py]: Agreed. Does this block this PR? I think this issue is meant to address this concern https://github.com/gitcoinco/web/issues/3101

**Pull request merged**